platform/darwin/src/MGLLocationManager.h

```diff
@@ -0,0 +1,175 @@
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MGLLocationManagerDelegate;
+
+/**
+ The `MGLLocationManager` protocol defines a set of methods that a class must
+ implement in order to serve as the location manager of an `MGLMapView`. A location
+ manager is responsible for notifying the map view about location-related events,
+ such as a change in the user’s location. This protocol is similar to the
+ Core Location framework’s `CLLocationManager` class, but your implementation
+ does not need to be based on `CLLocationManager`.
+ 
+ */
+@protocol MGLLocationManager <NSObject>
+
+@optional
+
+#pragma mark Determining Location Updates Accuracy
+
+/**
+ Specifies the minimum distance (measured in meters) a device must move horizontally
+ before a location update is generated.
+ 
+ The default value of this property is `kCLDistanceFilterNone` when `MGLMapView` uses  its
+ default location manager.
+ */
+@property(nonatomic, assign) CLLocationDistance distanceFilter;
+
+/**
+ Specifies the accuracy of the location data.
+ 
+ The default value is `kCLLocationAccuracyBest` when `MGLMapView` uses  its
+ default location manager.
+ 
+ @note Determining a location with greater accuracy requires more time and more power.
+ */
+@property (nonatomic, assign) CLLocationAccuracy desiredAccuracy;
+
+/**
+ Specifies the type of user activity associated with the location updates.
+ 
+ The location manager uses this property as a cue to determine when location updates
+ may be automatically paused.
+ 
+ The default value is `CLActivityTypeOther` when `MGLMapView` uses  its
+ default location manager.
+ */
+@property (nonatomic, assign) CLActivityType activityType;
+
+@required
+
+/**
+ The delegate to receive location updates.
+ 
+ Do not set the location manager’s delegate yourself. `MGLMapView` sets this property
+ after the location manager becomes `MGLMapView`’s location manager.
+ */
+@property (nonatomic, weak) id<MGLLocationManagerDelegate> delegate;
+
+#pragma mark Requesting Authorization for Location Services
+
+/**
+ Returns the current localization authorization status.
+ 
+ @see `+[CLLocationManger authorizationStatus]`
```

Comment:
This is a valuable note — we should do this with other properties that have `CLLocationManager` analogues with good external documentation that detail significant caveats, like `desiredAccuracy`, `distanceFilter`, `requestWhenInUseAuthorization`, etc.